asterixdb/asterix-algebra/src/main/java/org/apache/asterix/compiler/provider/DefaultRuleSetFactory.java

```diff
@@ -63,9 +63,15 @@
         defaultLogicalRewrites.add(new Pair<>(seqCtrlFullDfs, RuleCollections.buildLoadFieldsRuleCollection(appCtx)));
         defaultLogicalRewrites
                 .add(new Pair<>(seqCtrlFullDfs, RuleCollections.buildNormalizationRuleCollection(appCtx)));
+        defaultLogicalRewrites.add(new Pair<>(seqCtrlFullDfs, RuleCollections.buildLoadFieldsRuleCollection(appCtx)));
```

Comment:
Why do we need to add new collections of rules? Does your change need to be in multiple rules?

Reply:
Yes, overall we added 2 new rules for spatial join queries. The first rule adds a function that compute the MBR of the spatial field from the join inputs. The second rule manipulates the existing query plan to make the specific query plan for spatial join: adds SpatialTile Unnest operator to compute MBR's tile IDs, set the physical join operator to SpatialJoinPOperator, etc.